libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -1416,7 +1427,15 @@ class AnswerWithJustification(BaseModel):
                     key_name=tool_name, first_tool_only=True
                 )
         elif method == "json_mode":
-            llm = self.bind(response_format={"type": "json_object"})
+            if tools:
```

Comment:
seems sensible to me to enforce strict when passing tools here

Reply:
<img width="1260" alt="스크린샷 2024-10-09 오후 2 45 00" src="https://github.com/user-attachments/assets/ae16a274-d49f-443b-a9a2-67935a31eef0">
As you can see here, when use `json_mode`, tools must be strict by langchain rule